File: 1060-missing-element-in-sorted-array/1060-missing-element-in-sorted-array.cpp
class Solution {
public:
    int missingElement(vector<int>& nums, int k) {
        int l = 0, r = nums.size()-1;
        while(l < r) {
            int mid = r-(r-l)/2;
            if(nums[mid]-nums[0]-mid < k) l = mid;
            else r = mid-1;
        }
        return nums[0]+l+k;
    }
};

File: 1060-missing-element-in-sorted-array/1060-missing-element-in-sorted-array.py
class Solution(object):
    def missingElement(self, nums, k):
        l,r = 0,len(nums)-1
        while l < r:
            m = r-(r-l)//2
            if nums[m]-nums[0]-m < k: l = m
            else: r = m-1
        return nums[0]+l+k
        